scripts/ci/jenkins_pipeline_csi

```diff
@@ -25,9 +33,47 @@ pipeline {
                 sh './scripts/run_yamlcheck.sh'
             }
         }
-        stage ('CSI-controller & node: Build and push images') {
+        stage ('Build docker images') {
+            steps {
+                script {
+                    sh 'bash ./scripts/ci/run_variables_verification.sh'
+                    branch = sh(script:"echo $GIT_BRANCH | sed 's|/|.|g'",returnStdout:true).trim()
+                    specific_tag = "${IMAGE_VERSION}_b${BUILD_NUMBER}_${branch}"
+                    if (latest_tag.contains(GIT_BRANCH)) {
+                        controller_image_latest = docker.build("${CSI_CONTROLLER_IMAGE}", "--build-arg VERSION=${IMAGE_VERSION} --build-arg BUILD_NUMBER=${BUILD_NUMBER} -f ${Dockerfile_controller} .")
+                        node_image_latest = docker.build("${DOCKER_REGISTRY}/${CSI_NODE_IMAGE}", "--build-arg VERSION=${IMAGE_VERSION} --build-arg BUILD_NUMBER=${BUILD_NUMBER} -f ${Dockerfile_node} .")
+                    }
+                    controller_image_with_generated_tag = docker.build("${DOCKER_REGISTRY}/${CSI_CONTROLLER_IMAGE}:$specific_tag", "--build-arg VERSION=${IMAGE_VERSION} --build-arg BUILD_NUMBER=${BUILD_NUMBER} -f ${Dockerfile_controller} .")
+                    node_image_with_generated_tag = docker.build("${DOCKER_REGISTRY}/${CSI_NODE_IMAGE}:$specific_tag", "--build-arg VERSION=${IMAGE_VERSION} --build-arg BUILD_NUMBER=${BUILD_NUMBER} -f ${Dockerfile_node} .")
+                }
+            }
+        }
+        stage ('push docker images') {
             steps {
-                sh './scripts/ci/build_push_images.sh build/reports/images_url'
+                script {
+                    registryUrl = "${registry_protocol}://${DOCKER_REGISTRY}"
+                    docker.withRegistry(registryUrl, registryCredential ) {
+                        if (latest_tag.contains(GIT_BRANCH)) {
+                            controller_image_latest.push()
```

Comment:
if the `latest` tags push fails, will the stage stop and exit with failure? because I think the original code avoided that on purpose

Reply:
In the original code, the run will be green even If the latest tags push will fail or any push or build as long as the last command will succeded.

Reply:
I think you are wrong - not **any** push or build fail will result in a green run, only `latest` (or failure to save tags to file).
could you prove your statement? what makes you think that?

Reply:
because in bash script it doesn't stop if there is an error in one of the lines. In addition, the bash script will exit with exit code 0 if the last command of the script succeeded even when all other commands in the script failed.
To prove it I run a test pipeline that there are two Docker build commands in it' the first one will fail and the last one will succeed. The pipeline resulted in green.

Reply:
let me know when the proof is ready for me to see

Reply:
I was wrong, I work now with the original bash file